src/Service/ServiceFactory.php

```diff
@@ -172,6 +172,19 @@ public function getProgrammesService(): ProgrammesService
         return $this->instances['ProgrammesService'];
     }
 
+    public function getPromotionsService(): PromotionsService
+    {
+        if (!isset($this->instances['PromotionsService'])) {
```

Comment:
'ProgrammesService' is sort of a magic string that keeps getting written. Should it be a variable?